packages/malloy-interfaces/thrift/malloy.thrift

```diff
@@ -5,54 +5,53 @@
  * LICENSE file in the root directory of this source tree.
  */
 
+// In general, `...Info` structs represent "information about" an entity,
+// but not necessarily the _definition_ of that entity. E.g. a `Query` represents
+// an actual `query` object which can be serialized into Malloy code, compiled,
+// and run, whereas a `QueryInfo` represents the presence of a query in a model,
+// its name, its schema, and any other relevant information about it. A `QueryInfo`
+// may have a `definition`, which indicates that it is a query that is representable
+// as a `Query`. Not all queries are representable as such, because this interface
+// has more limited capabilities than the full Malloy language (e.g. index queries
+// are not representable here).
+
 struct ModelInfo {
-  1: required list<ModelEntryValue> entries,
-  2: optional list<Annotation> annotations,
-  3: required list<AnonymousQueryInfo> anonymous_queries,
+  1: required list<ModelEntry> entries;
```

Comment:
`required` is deprecated, but is needed to generate typescript types which accurately reflect which fields are _supposed to be present_. @narreola Is there a way you can think of for us to annotate fields as required such that we can generate TS types properly, but without actually making them required in the thrift?